feature/metis/conversation/src/test/kotlin/de/tum/informatics/www1/artemis/native_app/feature/metis/conversation/service/storage/impl/MetisStorageServiceImplTest.kt

```diff
@@ -0,0 +1,171 @@
+class MetisStorageServiceImplTest {
+
+    private val databaseProviderMock = MetisDatabaseProviderMock(InstrumentationRegistry.getInstrumentation().context)
+    private val sut = MetisStorageServiceImpl(databaseProviderMock)
+
+    private val host = "host"
+
+    private val author = User(id = 20, name = "AuthorName")
+    private val parentClientPostId = "parent-client-id-0"
+    private val answerClientPostId = "answer-client-id-0"
+
+    private val course: MetisContext.Course = MetisContext.Course(courseId = 1)
+    private val conversation = OneToOneChat(id = 2)
+    private val metisContext = MetisContext.Conversation(course.courseId, conversation.id)
+
+    private val localAnswerPojo = AnswerPostPojo(
+        parentPostId = parentClientPostId,
+        postId = answerClientPostId,
+        resolvesPost = false,
+        basePostingCache = AnswerPostPojo.BasePostingCache(
+            serverPostId = 0,
+            authorId = author.id,
+            creationDate = Clock.System.now(),
+            updatedDate = null,
+            content = "Answer post content 0",
+            authorRole = UserRole.USER,
+            authorName = author.name!!
+        ),
+        reactions = emptyList(),
+        serverPostIdCache = AnswerPostPojo.ServerPostIdCache(
+            serverPostId = null     // Only local answer post, no server id
+        )
+    )
+
+    private val basePostPojo = PostPojo(
+        clientPostId = parentClientPostId,
+        serverPostId = 0,
+        content = "Base post content",
+        resolved = false,
+        updatedDate = null,
+        creationDate = Clock.System.now(),
+        authorId = author.id,
+        title = null,
+        authorName = author.name!!,
+        authorRole = UserRole.USER,
+        courseWideContext = null,
+        tags = emptyList(),
+        answers = emptyList(),
+        reactions = emptyList()
+    )
+
+    private val basePost = StandalonePost(basePostPojo, conversation)
+    private val localAnswer = AnswerPost(localAnswerPojo, basePost)
+
+    @Test
+    fun testInsertClientSidePost() = runTest {
+        // GIVEN
+        sut.insertOrUpdatePosts(
+            host = host,
+            metisContext = metisContext,
+            posts = listOf(basePost),
+        )
+
+        // WHEN
+        sut.insertClientSidePost(
+            host = host,
+            metisContext = metisContext,
+            post = localAnswer,
+            clientSidePostId = answerClientPostId
+        )
+
+        // THEN
+        val posts = sut.getStoredPosts(
+            serverId = host,
+            metisContext = metisContext
+        ).loadAsList()
+
+        assertEquals(1, posts.size)
+        assertEquals(1, posts.first().answers.size)
+    }
+
+
+    @Test
+    fun testUpgradeClientSideAnswerPost() = runTest {
+        // Okay, this is super weird. Somehow the test sql database has problems if the test method
+        // name is too long. That's why I had to create this delegation method.
```

Comment:
I'm not sure but in this particular case I would maybe even accept a less descriptive function title, even if it is not that straight forward. But in my opinion it's better than a weird delegation. What do you guys think @TimOrtel and @FelberMartin?

Reply:
Fair point, I could add the more descriptive parts to the respective comments ( `// GIVEN, WHEN, THEN`)